### ISSUE TITLE
Error whilst trying the example  the trait `AsRef<[_]>` is not implemented

### ISSUE THREAD

**Issue by wlwatkins** [espdev/csaps-rs#3]
I cannot run the example, I surely did something wrong, but the `&x`, `&y` and `&w` are underlined basically saying the AsRef trait is not implemented?Cargo.toml:
ndarray = {version = "0.15.6", feature =["rayon"]}
ndarray-linalg = { version = "0.16.0", features = ["intel-mkl-static"] }
csaps = "0.3.0"

```rust
// some of the imports are not used
use csaps::CubicSmoothingSpline;
use ndarray::{array, Array1, Array2, Zip, Axis, ArrayView2, ArrayView1, stack, Array, Dim, ArrayViewMut};
    let x = array![1.0, 2.0, 3.0, 4.0];
    let y = array![0.5, 1.2, 3.4, 2.5];
    let w = array![1.0, 0.7, 0.5, 1.0];
    let smooth = 0.85;

    let s = CubicSmoothingSpline::new(&x, &y)
        .with_weights(&w)
        .with_smooth(smooth)
        .make().unwrap();
```

```
error[E0277]: the trait bound `ArrayBase<OwnedRepr<{float}>, Dim<[usize; 1]>>: AsRef<[_]>` is not satisfied
   --> src\wmap\apply.rs:61:39
    |
61  |     let s = CubicSmoothingSpline::new(&x, &y)
    |             ------------------------- ^^ the trait `AsRef<[_]>` is not implemented for `ArrayBase<OwnedRepr<{float}>, Dim<[usize; 1]>>`
    |             |
    |             required by a bound introduced by this call
    |
    = note: required for `ndarray::ArrayBase<ndarray::ViewRepr<&_>, ndarray::dimension::dim::Dim<[usize; 1]>>` to implement `From<&ArrayBase<OwnedRepr<{float}>, Dim<[usize; 1]>>>`
    = note: required for `&ArrayBase<OwnedRepr<{float}>, Dim<[usize; 1]>>` to implement `Into<ndarray::ArrayBase<ndarray::ViewRepr<&_>, ndarray::dimension::dim::Dim<[usize; 1]>>>`
    = note: required for `&ArrayBase<OwnedRepr<{float}>, Dim<[usize; 1]>>` to implement `ndarray::arraytraits::AsArray<'_, _>`
note: required by a bound in `CubicSmoothingSpline::<'a, T, D>::new`
```

**Comment by willmendil** [comment]
Ok so apparently it is because csaps uses ndarray 0.13, and now we are at 0.15 with breaking changes. 
Are there any plans of updating the crate?

**Comment by espdev** [comment]
Refers to: https://github.com/espdev/csaps-rs/pull/2

**Comment by espdev** [comment]
Hi @willmendil

I'm sorry, I don't have time to deal with this package right now. Any help and PR are welcome.

**Comment by willmendil** [comment]
Thanks for the insight. I'll take a look, but I don't think I have the skills to fix it.

**Comment by espdev** [comment]
Seems fixed in #2 and #4
